solidity/contracts/KeepRandomBeaconOperator.sol

```diff
@@ -147,6 +147,8 @@ contract KeepRandomBeaconOperator is ReentrancyGuard {
         // groups expired, we first want to make a cleanup.
         groups.expireOldGroups();
         require(numberOfGroups() == 0, "Groups exist");
+        // Executing candidate cleanup in case of previous genesis failure.
+        groupSelection.cleanupCandidates();
```

Comment:
Is there a reason why do we call only `clenupCandidates()` and not the full cleanup? Cleaning up just candidates is probably enough but executing the full cleanup could be less confusing. This could also be an opportunity to make compiler happier and eliminate:
```
    function stop(Storage storage self) public {
    ^ (Relevant source part starts here and spans across multiple lines).
,/Users/piotr/go/src/github.com/keep-network/keep-core/solidity/contracts/libraries/operator/GroupSelection.sol:100:5: Warning: Variable is shadowed in inline assembly by an instruction of the same name
    function stop(Storage storage self) public {
```

We could rename `groupSelection.stop()` to e.g. `groupSelection.finish()` and call the entire `groupSelection.finish()` (which is already `public`) here.

What do you think?

Reply:
> Is there a reason why do we call only clenupCandidates() and not the full cleanup?

In current implementation for genesis we do the full cleanup, but it is broken down to 2 pieces:
- candidates cleanup in `genesis()`
- tickets cleanup in `genesis -> startGroupSelection -> groupSelection.start()`

> We could rename groupSelection.stop() to e.g. groupSelection.finish() and call the entire groupSelection.finish() 

Calling `groupSelection.finish()` from `genesis()`? If yes, then we would need to refactor code so we won't call cleanupTickets 2 times. But at the same time, we execute `cleanupTickets()` every time (createGroup()) when DKG fee pool has enough eth.

Reply:
Yeah, for sure we can'd change anything in `startGroupSelection` because we'll run into gas estimate problems again. We need to limit ourselves to changes in `genesis`.

I think it's not that bad if we `cleanupTickets` twice. It's not an expensive function and I think we can sacrifice just a little more gas during `genesis` for a cleaner code. I think we should have:

```
groupSelection.finish() // cleanup after potential failed DKG
startGroupSelection(...)
```

IMO it's a little cleaner compare to what we have now. What do you think?

Reply:
If it's not bad calling `cleanupTickets` twice, then sure, that was my only concern.

Reply:
Yeah, I think in all other cases but genesis I'd say it's not the best solution. In this single case, I think we are fine making this tradeoff.